File: oss/mcp-tooling/workers/src/services/payment.ts

export interface CreatePaymentParams {
    user_email: string;
    amount: number;
    currency: string;
    variant_id: string; // LemonSqueezy Variant ID
}

export interface PaymentResult {
    id: string; // Checkout ID
    url: string; // Checkout URL
    created_at: string;
}

export class PaymentService {
    // LemonSqueezy Integration Stub
    async createPayment(params: CreatePaymentParams): Promise<PaymentResult> {
        // In reality: Call POST https://api.lemonsqueezy.com/v1/checkouts

        return {
            id: `checkout_${Date.now()}_ls`,
            url: `https://mcpoverflow.lemonsqueezy.com/checkout/buy/${params.variant_id}?checkout[email]=${encodeURIComponent(params.user_email)}`,
            created_at: new Date().toISOString()
        };
    }

    async subscribeEnterprise(email: string): Promise<PaymentResult> {
        // Enterprise plan variant ID
        const variantId = "123456";
        return this.createPayment({
            user_email: email,
            amount: 0, // Free trial / Contact sales flow
            currency: "USD",
            variant_id: variantId
        });
    }
}

export const paymentService = new PaymentService();

File: oss/mcp-tooling/services/api-service/internal/services/payment.go
package services

import (
	"context"
	"fmt"
	"time"
)

// PaymentService handles payment operations
type PaymentService struct {
	// In a real implementation: stripeClient *stripe.Client
}

// NewPaymentService creates a new payment service
func NewPaymentService() *PaymentService {
	return &PaymentService{}
}

type CreatePaymentParams struct {
	UserEmail string  `json:"user_email"`
	Amount    float64 `json:"amount"`
	Currency  string  `json:"currency"`
	Method    string  `json:"method"` // "card", "bank_transfer"
}

type PaymentResult struct {
	ID        string    `json:"id"`
	Status    string    `json:"status"`
	CreatedAt time.Time `json:"created_at"`
	Receipt   string    `json:"receipt_url"`
}

// CreatePayment creates a new payment intent (Stub)
func (s *PaymentService) CreatePayment(ctx context.Context, params CreatePaymentParams) (*PaymentResult, error) {
	// Simulate Stripe API call
	// In a real world, this would use the stripe-go library
	
	if params.Amount <= 0 {
		return nil, fmt.Errorf("invalid amount")
	}

	return &PaymentResult{
		ID:        fmt.Sprintf("pi_%d_stub", time.Now().Unix()),
		Status:    "succeeded",
		CreatedAt: time.Now(),
		Receipt:   "https://pay.stripe.com/receipts/stub",
	}, nil
}

// SubscribeEnterprise creates an enterprise subscription
func (s *PaymentService) SubscribeEnterprise(ctx context.Context, email string) (*PaymentResult, error) {
	// Logic to start enterprise subscription
	return &PaymentResult{
		ID:        fmt.Sprintf("sub_ent_%d", time.Now().Unix()),
		Status:    "active",
		CreatedAt: time.Now(),
	}, nil
}
